credentials/credentials.go

```diff
@@ -193,6 +194,31 @@ func RequestInfoFromContext(ctx context.Context) (ri RequestInfo, ok bool) {
 	return
 }
 
+// AddressInfo contains address related data attached to the context passed to
+// ClientHandshake. This makes it possible for balancer implementations to pass
```

Comment:
As before, "balancer, resolver, grpc, etc".  There's a flow of data here that the balancer and handshaker are only parts of.

Reply:
Done.